src/alfasim_sdk/_internal/alfacase/alfacase_to_case.py

```diff
@@ -322,30 +322,48 @@ def get_dict_of_arrays_loader(
     )
 
 
-def load_curve(key: str, alfacase_content: DescriptionDocument, category: str) -> Curve:
+def load_curve(
+    key: str,
+    alfacase_content: DescriptionDocument,
+    category: str,
+    domain_category: str = "time",
+) -> Curve:
     """
     Create a barril.curve.curve.Curve instance from the given YAML content.
     # TODO: ASIM-3556: All atributes from this module should get the category from the CaseDescription
     """
     curve_in_alfacase = alfacase_content[key]
     return Curve(
         load_array("image", curve_in_alfacase, category),
-        load_array("domain", curve_in_alfacase, "time"),
+        load_array("domain", curve_in_alfacase, domain_category),
     )
 
 
 @lru_cache(maxsize=None)
 def get_curve_loader(
-    *, category: Optional[str] = None, from_unit: Optional[str] = None
+    *,
+    category: Optional[str] = None,
+    from_unit: Optional[str] = None,
+    domain_category: Optional[str] = None,
+    from_domain_unit: Optional[str] = None,
 ) -> Callable:
     """
     Return a load_curve function pre-populated with the category
 
     If ``from_unit`` is provided, the category parameter will be filled with
     the default category for the given unit.
     """
+    if domain_category is not None or from_domain_unit is not None:
```

Comment:
Perhaps add an `assert` to ensure that both `domain_category` and `from_domain_unit` aren't `None`

Reply:
Sure

Reply:
`_obtain_category_for_scalar` already check for that so it is not needed.
```
def _obtain_category_for_scalar(category: str, from_unit: str) -> str:
    if category is not None and from_unit is not None:
        raise ValueError(
            "Both parameters 'category' and 'from_unit' were provided, only one must be informed"
        )
    if category is None and from_unit is None:
        raise ValueError("Either 'category' or 'from_unit' parameter must be defined")

    return category or get_category_for(from_unit)
```

Reply:
Hum, good to know. I will remove this check then